clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -3443,9 +3440,27 @@ class SyclKernelIntHeaderCreator : public SyclKernelFieldHandler {
   }
 
   bool handlePointerType(FieldDecl *FD, QualType FieldTy) final {
-    addParam(FD, FieldTy,
-             ((StructDepth) ? SYCLIntegrationHeader::kind_std_layout
-                            : SYCLIntegrationHeader::kind_pointer));
+    QualType PointeeTy = FieldTy->getPointeeType();
+    unsigned Dim = 0;
+    ASTContext &Context = SemaRef.getASTContext();
+    // VLAs are encoded as (Dim << 24) | Size, where Dim is the
+    // number of VLA dimensions and Size is the size of the
+    // underlying type. Dim indicates that an equal number of
+    // preceding entries (of std_layout kind) describe the
```

Comment:
Is this actually sufficient though? As best I can tell, `std_layout` kinds eventually go through `addParam()`, which only specifies the offset and the size of the dimension, but not what variable it's tracked from. e.g.,
```
int x = 10;
x++;
x += 10;
int vla[x]; // x == 21
```
I don't see how the `x` would be tracked through such that a `sizeof(vla)` on the device side would give the correct answer.